lib/internal/streams/end-of-stream.js

```diff
@@ -199,6 +205,20 @@ function eos(stream, options, callback) {
     });
   }
 
+  if (options.signal && !closed) {
+    const abort = () => callback(new AbortError());
+    if (options.signal.aborted) {
+      process.nextTick(abort);
+    } else {
+      const originalCallback = callback;
+      callback = once((...args) => {
+        options.signal.removeEventListener('abort', abort);
+        originalCallback(...args);
```

Comment:
You can avoid unsafe array iteration here.
```suggestion
        ReflectApply(originalCallback, null, args);
```

Reply:
done